WordPressKit/StatsServiceRemoteV2.swift

```diff
@@ -0,0 +1,193 @@
+    // so we do this "fun" dance in a separate method.
+    public func getInsight(completion: @escaping ((StatsLastPostInsight?, Error?) -> Void)) {
+         getLastPostInsight(completion: completion)
+    }
+
+    private func getLastPostInsight(completion: @escaping ((StatsLastPostInsight?, Error?) -> Void)) {
+        let properties = StatsLastPostInsight.queryProperties
+        let pathComponent = StatsLastPostInsight.pathComponent
+
+        let path = self.path(forEndpoint: "sites/\(siteID)/\(pathComponent)", withVersion: ._1_1)
+
+        wordPressComRestApi.GET(path, parameters: properties, success: { (response, _) in
+            guard
+                let jsonResponse = response as? [String: AnyObject],
+                let posts = jsonResponse["posts"] as? [[String: AnyObject]],
+                let post = posts.first,
+                let postID = post["ID"] as? Int else {
+                    completion(nil, ResponseError.decodingFailure)
+                    return
+            }
+
+            self.getPostViews(for: postID) { (views, error) in
+                guard
+                    let views = views,
+                    let insight = StatsLastPostInsight(jsonDictionary: post, views: views) else {
+                        completion(nil, ResponseError.decodingFailure)
+                        return
+
+                }
+
+                completion(insight, nil)
+            }
+        }, failure: {(error, _) in
+            completion(nil, error)
+        })
+    }
+
+    public func getPostViews(`for` postID: Int, completion: @escaping ((Int?, Error?) -> Void)) {
+        let parameters = ["fields": "views" as AnyObject]
+
+        let path = self.path(forEndpoint: "sites/\(siteID)/stats/post/\(postID)", withVersion: ._1_1)
+
+        wordPressComRestApi.GET(path,
+                                parameters: parameters,
+                                success: { (response, _) in
+                                    guard
+                                        let jsonResponse = response as? [String: AnyObject],
+                                        let views = jsonResponse["views"] as? Int else {
+                                            completion(nil, ResponseError.decodingFailure)
+                                            return
+                                    }
+                                    completion(views, nil)
+                                }, failure:  { (error, _) in
+                                    completion(nil, error)
+                                }
+        )
+    }
+
+}
+
+// This serves both as a way to get the query properties in a "nice" way,
+// but also as a way to narrow down the generic type in `getInisght:` method.
+public protocol InsightProtocol {
+    static var queryProperties: [String: AnyObject] { get }
+    static var pathComponent: String { get }
+
+    init?(jsonDictionary: [String: AnyObject])
```

Comment:
How would you feel about having your implementors conform to `Decodable`?

Reply:
I desperately wish I could just use Decodable for this, but the underlying layer of `WordPressComRestApi` already parses the response from `Data` into a JSON dictionary, and `Codable` doesn't have a dictionary `Decoder` :(